### DISCUSSION TITLE
No entrypoint file found && failed to create new program: No such project

### DISCUSSION THREAD

**Discussion by serdarakyol** [deta/cloud-docs#701] [Help] [answered]
Hello, I just discovered DETA. I was trying to follow the FastAPI documentation. Created an account, installed deta cli, and login successfully. But I failed on the `deta new` command. I had 2 situation
# First situation
I have the folder structure as below

![image](https://user-images.githubusercontent.com/36725883/209968349-b6754188-387a-42c8-9ac2-4ba987c8aead.png)

When I run `deta new`, I got `No entrypoint file found in '/path/sabiduria-tool'. Please, provide a name or path to create a new micro elsewhere. See "deta new --help".'` error.

# Second situation
Changed folder structure as below

![image](https://user-images.githubusercontent.com/36725883/209968419-15b14087-9780-4f97-a414-89c962e8c79d.png)

And got this error: `Error: failed to create new program: No such project`

Here is my main file
```python
from fastapi import FastAPI

from sabiduria_tool_api.core.config import API_PREFIX, APP_NAME, APP_VERSION, IS_DEBUG
from sabiduria_tool_api.core.event_handler import start_app_handler, stop_app_handler
from sabiduria_tool_api.api.routes.router_sabiduria_tool import router

def get_app() -> FastAPI:
    fast_app = FastAPI(title=APP_NAME, version=APP_VERSION, debug=IS_DEBUG)
    fast_app.include_router(router, prefix=API_PREFIX)

    fast_app.add_event_handler("startup", start_app_handler(fast_app))
    fast_app.add_event_handler("shutdown", stop_app_handler(fast_app))

    return fast_app


app = get_app()

```
I used the app variable as the documentation mentioned. Can anyone tell me what is wrong here, please?

**Comment by sudo-subg** [comment, marked as answer]
https://docs.deta.sh/docs/cli/commands/#notes
> If path is not an empty directory and does not have an entrypoint file (either main.py or index.js) a name must be provided, under which deta will create a micro with a starter app.

So your first situation showed `No entrypoint file found in '/path/sabiduria-tool'. Please, provide a name or path to create a new micro elsewhere. See "deta new --help".'` cause it's missing the entrypoint file (main.py) in the root `/path/sabiduria-tool/` path. In your case you need to run `deta  new` in the  `/path/sabiduria-tool/sabiduria_tool_api` directory.

And for the second situation iirc new users need to create their first project at [deta dashboard](https://web.deta.sh/) after creating an account.

**Reply by serdarakyol** [reply]
I have tested that before. When I run `deta new` inside the `/path/sabiduria-tool/sabiduria_tool_api`, I am getting `Error: failed to create new program: No such project` error. 
When I run inside `/path/sabiduria-tool`, then I am getting `No entrypoint file found in '/path/sabiduria-tool'. Please, provide a name or path to create a new micro elsewhere. See "deta new --help".'`

**Reply by sudo-subg** [reply]
As I said before;
> And for the second situation iirc new users need to create their first project at [deta dashboard](https://web.deta.sh/) after creating an account.

Can you try this once and check if your first project is created or not.

**Reply by serdarakyol** [reply]
Oh my bad. You were right. I thought `deta new` will create that automatically.
